test/e2e/placement_selecting_resources_test.go

```diff
@@ -483,7 +483,7 @@ var _ = Describe("validating CRP when adding resources in a matching namespace",
 			workResourcesPlacedActual := func() error {
 				return validateWorkNamespaceOnCluster(memberCluster, workNamespaceName)
 			}
-			Expect(workResourcesPlacedActual()).Should(Succeed(), "Failed to place work namespace %s on member cluster %s", workNamespaceName, memberCluster.ClusterName)
+			Eventually(workResourcesPlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place work namespace %s on member cluster %s", workNamespaceName, memberCluster.ClusterName)
```

Comment:
are you sure this test is flaky?

here it should use expect, cause crp has already applied the resources.
no need to use eventually.

Reply:
it failed in these two commits:
- [feat: property-based scheduling: enable the property-based scheduling experience #2695](https://github.com/Azure/fleet/actions/runs/8506422376/job/23296596391)
- [feat: apply overrides before creating the work #2705](https://github.com/Azure/fleet/actions/runs/8517373471/job/23327835302)

Reply:
let me investigate it, this one looks like a bug.